### ISSUE TITLE
nemo2riva of tts_en_fastpitch:1.4.0 fails with CUDA out of memory

### ISSUE THREAD

**Issue by itzsimpl** [NVIDIA/NeMo#4012]
@borisfom As per comment https://github.com/NVIDIA/NeMo/issues/3619#issuecomment-1089551439, I'm re-opening the bug as the issue persists with `nemo:22.01` (1.7.1), `nemo:1.7.2` and `nemo:1.9.0rc0` (e4ee26b601a8e6f907f926019367d3d545c62f11) combined with `Riva:2.0.0`. I tried both by starting off with `nemo:22.01` and upgrading to more recent versions of nemo, as well as building new containers from the specific GitHub repo tag. I tested with T4 (16GB) and A100 (40GB) GPUs. On all occasions the conversion of [tts_en_fastpitch_align.nemo](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/tts_en_fastpitch) fails with `RuntimeError: CUDA out of memory`.

**Comment by itzsimpl** [comment]
A little bit of update. The most recent version of nemo I could convert [tts_en_fastpitch_align.nemo](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/tts_en_fastpitch) was `nemo:1.5.1`. Reviewing the changes `1.6.2` brought, the offending change seems to be in 
https://github.com/NVIDIA/NeMo/blob/7da391634e276df7273b4da1443aa1c5b3221a32/nemo/collections/tts/models/fastpitch.py#L473-L494
Changing L473 to
```python
     def input_example(self, max_batch=1, max_dim=44): 
```
the values that were hardcoded in `1.5.1` resolves the issue for [tts_en_fastpitch_align.nemo](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/tts_en_fastpitch). I can even export a model that was built on `nemo:1.8.1`. However, if I apply the same fix in `nemo:1.8.1` and try to convert a model built with `nemo:1.8.1`, the conversion fails with the following trace
```python
...
[NeMo I 2022-04-25 13:35:26 features:259] PADDING: 1
[NeMo I 2022-04-25 13:35:26 features:276] STFT using torch
[NeMo I 2022-04-25 13:35:28 save_restore_connector:209] Model FastPitchModel was successfully restored from /data/data/nemo/fastpitch.nemo.
sz=(1, 44)
[NeMo I 2022-04-25 13:35:29 export_utils:261] Swapped 28 modules
Warning: Constant folding - Only steps=1 can be constant folded for opset >= 10 onnx::Slice op. Constant folding not applied.
Warning: Constant folding - Only steps=1 can be constant folded for opset >= 10 onnx::Slice op. Constant folding not applied.
Warning: Constant folding - Only steps=1 can be constant folded for opset >= 10 onnx::Slice op. Constant folding not applied.
WARNING: Logging before flag parsing goes to stderr.
E0425 13:35:45.721092 140244939851584 cookbook.py:98] Export failed. Please make sure your NeMo model class (nemo.collections.tts.models.fastpitch.FastPitchModel) has working export() and that you have the latest NeMo package installed with [all] dependencies.
Traceback (most recent call last):
  File "/opt/conda/bin/nemo2riva", line 8, in <module>
    sys.exit(nemo2riva())
  File "/opt/conda/lib/python3.8/site-packages/nemo2riva/cli/nemo2riva.py", line 50, in nemo2riva
    Nemo2Riva(args)
  File "/opt/conda/lib/python3.8/site-packages/nemo2riva/convert.py", line 68, in Nemo2Riva
    save_archive(model=model, save_path=riva_out, cfg=cfg, artifacts=artifacts, metadata=manifest['metadata'])
  File "/opt/conda/lib/python3.8/site-packages/nemo2riva/cookbook.py", line 103, in save_archive
    raise e
  File "/opt/conda/lib/python3.8/site-packages/nemo2riva/cookbook.py", line 83, in save_archive
    _, descriptions = model.export(
  File "/nemo/nemo/core/classes/exportable.py", line 142, in export
    torch.onnx.export(
  File "/opt/conda/lib/python3.8/site-packages/torch/onnx/__init__.py", line 319, in export
    return utils.export(model, args, f, export_params, verbose, training,
  File "/opt/conda/lib/python3.8/site-packages/torch/onnx/utils.py", line 113, in export
    _export(model, args, f, export_params, verbose, training, input_names, output_names,
  File "/opt/conda/lib/python3.8/site-packages/torch/onnx/utils.py", line 734, in _export
    params_dict = torch._C._jit_pass_onnx_deduplicate_initializers(graph, params_dict,
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument other in method wrapper__equal)
```

**Comment by itzsimpl** [comment]
Further update. I can confirm that with `pytorch:22.04-py3` the error
```python3
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument other in method wrapper__equal)
```
is resolved. 

Tested with `nemo:1.8.1`, `nemo:1.8.2` and `nemo:1.9.0rc0`, all installed in `pytorch:22.04-py3`. In all cases conversion succeeds if one patches `FastPitchModel.input_example` to 
```python3 
def input_example(self, max_batch=1, max_dim=44)
```
or passes parameter `--max-dim 44` to `nemo2riva`.

**Comment by okuchaiev** [comment]
@borisfom  any thoughts on this?

**Comment by okuchaiev** [comment]
ping @borisfom

**Comment by borisfom** [comment]
And yes 'tensors on the same device' error was due to pytorch bug (fixed in 22.04).
